plugins/catalog-react/src/components/EntityGenericPicker/EntityGenericPicker.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Box,
+  Checkbox,
+  FormControlLabel,
+  TextField,
+  Typography,
+} from '@material-ui/core';
+import CheckBoxIcon from '@material-ui/icons/CheckBox';
+import CheckBoxOutlineBlankIcon from '@material-ui/icons/CheckBoxOutlineBlank';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import { Autocomplete } from '@material-ui/lab';
+import React, { useEffect, useMemo, useState } from 'react';
+import { useEntityList } from '../../hooks/useEntityListProvider';
+import { get } from 'lodash';
+import { Entity } from '@backstage/catalog-model';
+import { EntityGenericFilter } from '../../filters';
+
+/** @public */
+export type CatalogReactEntityGenericPickerClassKey = 'input';
+
+const icon = <CheckBoxOutlineBlankIcon fontSize="small" />;
+const checkedIcon = <CheckBoxIcon fontSize="small" />;
+
+/**
+ * Props for {@link EntityGenericPicker}.
+ *
+ * @public
+ */
+export interface EntityGenericPickerProps {
+  name: string;
+  filterValue: string;
+  filterKey?: string;
+}
+
+/** @public */
+export const EntityGenericPicker = (props: EntityGenericPickerProps) => {
```

Comment:
I feel it's important that we use the new facets API (#9491) in the catalog for this tbh, and not add any more client-side filtering and value discovery into the mix. We're moving towards full use of pagination for the catalog and related frontend components and this seems like it wouldn't be compatible with that in its current form.

I'm not sure entire where we're at with that work tbh though, @vinzscam, @backstage/catalog-core? Do you have any thoughts on this new filter component and whether we can build something now that's future compatible or maybe should hold off a bit?

Reply:
sorry I totally missed this PR. Yes, this doesn't go in the direction we would like to go and such component needs to be rebuilt in the future. Honestly I would hold off a bit, but in case this is really needed feel free to go on with it. Since the changes don't touch `CatalogPage`, `CatalogTable`, `EntityListProvider` I'm happy 😄.

Reply:
The changes don't touch these files. I understand that adding another filter component might add some extra work while re structuring all the filters to use the new facets api, in progress. Right now, we want to have the ability to add custom filter components for the Scaffolder Page. My changes to the scaffolder page lets the user to send any custom filter component as a property to the Scaffolder page and render them.